lib/remote/apilistener-configsync.cpp

```diff
@@ -393,6 +395,52 @@ void ApiListener::UpdateConfigObject(const ConfigObject::Ptr& object, const Mess
 	}
 }
 
+/**
+ * Syncs the specified object and its direct and intermediate parents to the provided client
+ * in topological order of their dependency graph recursively.
+ *
+ * Objects that the client does not have access to are skipped without going through their dependency graph.
+ *
+ * Please do not use this method to forward remote generated cluster updates; it should only be used to
+ * send local updates to that specific non-nullptr client.
+ *
+ * @param object ConfigObject::Ptr The config object you want to sync.
+ * @param client JsonRpcConnection::Ptr The JsonRpc client you send the update to.
+ * @param syncedObjects std::unordered_set<ConfigObject*> Used to cache the already synced objects.
+ */
+void ApiListener::UpdateConfigObjectWithParents(const ConfigObject::Ptr& object, const JsonRpcConnection::Ptr& client,
+	std::unordered_set<ConfigObject*>& syncedObjects)
+{
+	// Don't sync already synced objects.
+	if (syncedObjects.find(object.get()) != syncedObjects.end()) {
+		return;
+	}
+
+	VERIFY(client);
+
+	Endpoint::Ptr endpoint = client->GetEndpoint();
+	ASSERT(endpoint);
+
+	Zone::Ptr azone = endpoint->GetZone();
+
+	/* don't sync objects for non-matching parent-child zones */
+	if (!azone->CanAccessObject(object)) {
+		return;
+	}
+	syncedObjects.emplace(object.get());
+
+	for (const Object::Ptr& parent : DependencyGraph::GetParents(object)) {
```

Comment:
* FWIW, this just ignores dependency cycles. Which is ok , given that we don't forbid them. #10189